engine/gamesys/src/gamesys/components/comp_camera.cpp

```diff
@@ -36,84 +36,78 @@ namespace dmGameSystem
     using namespace dmVMath;
 
     const uint32_t MAX_COUNT = 64;
-    const uint8_t MAX_STACK_COUNT = 8;
 
     struct CameraWorld;
 
     struct CameraComponent
     {
         dmGameObject::HInstance m_Instance;
-        CameraWorld* m_World;
-        float m_AspectRatio;
-        float m_Fov;
-        float m_NearZ;
-        float m_FarZ;
-        float m_OrthographicZoom;
-        dmVMath::Matrix4 m_View;
-        dmVMath::Matrix4 m_Projection;
-        uint32_t m_AutoAspectRatio : 1;
-        uint32_t m_AddedToUpdate : 1;
-        uint32_t m_OrthographicProjection : 1;
-        uint16_t m_ComponentIndex;
+        dmRender::HRenderCamera m_RenderCamera;
+        CameraWorld*            m_World;
+        dmVMath::Matrix4        m_View;
+        dmVMath::Matrix4        m_Projection;
+        uint16_t                m_ComponentIndex;
+        uint8_t                 m_AddedToUpdate : 1;
     };
 
     struct CameraWorld
     {
-        dmArray<CameraComponent> m_Cameras;
-        dmArray<CameraComponent*> m_FocusStack;
+        dmArray<CameraComponent>  m_Cameras;
+        dmArray<CameraComponent*> m_CameraStack;
```

Comment:
We still use a camera stack because we keep track of a "main" camera based on the order for the stack. If the current main camera gets removed we need to know which was the previous main camera so we can put that into the camera.MAIN property